canister/src/lib.rs

```diff
@@ -38,6 +38,11 @@ use std::convert::TryInto;
 use std::{cell::RefCell, cmp::max};
 use utxo_set::UtxoSet;
 
+/// Used to determine if the canister is synced based on difference
+/// between the maximum height of all block headers and the
+/// maximum height of all unstable blocks
+pub const SYNCING_THRESHOLD: u32 = 2;
```

Comment:
nit: This isn't really a syncing threshold - we're always syncing no matter what. Instead, it's a threshold of whether or not we are synced.

Also, this constant should be private.

How about?

```suggestion
/// The maximum number of blocks the canister can be behind the tip to be considered synced.
const SYNCED_THRESHOLD: u32 = 2;
```